jujugui/static/gui/src/app/components/machine-view/machine-view.js

```diff
@@ -227,7 +229,32 @@ class MachineView extends React.Component {
     @param {String} id The machine id to select.
   */
   selectMachine(id) {
-    this.setState({selectedMachine: id});
+    this.props.changeState({gui: {machines: id}});
+  }
+
+  /**
+    Get the currently selected machine and container.
+    @returns {Object} The selected machine and container.
+  */
+  _getSelected() {
+    const machine = this.props.machine;
+    let selected = {
+      machine: null,
+      container: null
+    };
+    if (machine) {
+      // The ID will be in the format "machine-id/container-type/container-id"
```

Comment:
In the machine view we need to know what both the selected machine and selected container are at the same time. Unfortunately the parsing in `parseMachineName` returns that info in an inconsistent way. I think what I'll do is implement an intermediate method that normalises the results in the status view.